### ISSUE TITLE
Editor crash with .NET 8.0

### ISSUE THREAD

**Issue by 2dogtwo** [godotengine/godot#90299]
### Tested versions

Godot Engine v4.3.dev.mono.custom_build.29b3d9e9e (2024-03-28 18:06:55 UTC) - https://godotengine.org
OpenGL API 4.5 (Core Profile) Mesa 20.3.5 - Compatibility - Using Device: Mesa/X.org - llvmpipe (LLVM 11.0.1, 128 bits)

### System information

debian 11  arm64 opengl

### Issue description

。

### Steps to reproduce

I am trying to export Android APK. But when running on Android, you can only see the startup screen. Afterwards, the application will crash. So I want to try Gradle builds. But export compilation has just begun. The interface displays the downloaded file. The program crashes. Afterwards, no matter which project I run. Even if it's a new project. The editor will suddenly crash. Even if no action is taken. Before that. The program can be used normally. The running environment is to install Debian11 arm64 in the chroot container on the phone. The newly compiled version 4.3 initially had some issues. After adjusting the settings, it can be used normally.. I was able to use it normally from version 3.5 to version 4.0 before. What should I do now. I am trying to delete the data directory of Godot. But it has no effect. I have also tried restarting my phone. Debian in the container has not been restarted. I don't know how to operate. My English is not good. This is from online translation. I hope there are no errors

```
root@localhost:~# /root/godot/godot.mono --audio-driver 'Dummy' --disable-vsync
Godot Engine v4.3.dev.mono.custom_build.29b3d9e9e (2024-03-28 18:06:55 UTC) - https://godotengine.org
OpenGL API 4.5 (Core Profile) Mesa 20.3.5 - Compatibility - Using Device: Mesa/X.org - llvmpipe (LLVM 11.0.1, 128 bits)

Editing project: /root/ce
Godot Engine v4.3.dev.mono.custom_build.29b3d9e9e (2024-03-28 18:06:55 UTC) - https://godotengine.org
OpenGL API 4.5 (Core Profile) Mesa 20.3.5 - Compatibility - Using Device: Mesa/X.org - llvmpipe (LLVM 11.0.1, 128 bits)

ERROR: Condition "p_font.is_null()" is true. Returning: false
   at: add_string (scene/resources/text_paragraph.cpp:420)
Could not find version of build tools that matches Target SDK, using 33.0.2
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at System.Threading.ReaderWriterLockSlim.GetThreadRWCount(Boolean)
   at System.Threading.ReaderWriterLockSlim.TryEnterWriteLockCore(TimeoutTracker)
   at System.Threading.ReaderWriterLockSlim.EnterWriteLock()
   at System.Diagnostics.Process.OnSigChild(Int32, Int32)
```

### Minimal reproduction project (MRP)

  。。

**Comment by 2dogtwo** [comment]
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.beta.mono.custom_build (0ddd9c3e8f9088fa139e25b2903289d727073e12)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] linux-vdso.so.1(__kernel_rt_sigreturn+0) [0x78c6c2a6c0] (??:0)
-- END OF BACKTRACE --
 
I tried version 4.0. New error messages have been provided.. So is it true that there is a problem with the system? Can it be solved? To delete the container. Do you want to reinstall Debian?

**Comment by kitbdev** [comment]
> ERROR: Condition "p_font.is_null()" is true. Returning: false
   at: add_string (scene/resources/text_paragraph.cpp:420)

- Might be related to https://github.com/godotengine/godot/issues/90001
- Try checking if it is fixed by https://github.com/godotengine/godot/pull/90022

**Comment by 2dogtwo** [comment]
I have solved the problem. These issues caused by dotnet8.0.
I recompiled version 4.2.2 of Godot. But the problem remains. I suddenly thought that I had no problem using version 4.0 before. Why are there problems again now. I even reinstalled the system. It's impossible for problems to occur. So. The error was only caused by differences in software installed during compilation of version 4.3. And the error message is system XXXX, this is the content that only dotnet reports errors. So, I deleted all files and folders related to dotnet in the system, and then installed dotnet 6.0. This is the version I used for Godot 4.0. There are no issues with the test. Then install Dotnet7.0, which is required for exporting to Android. There are no issues after testing. Android APK can also be exported normally. So, this is an issue with DotNet 8.0. What are the issues with using Dotnet8.0 in Godot.. I see that others have also given feedback on the system XXXX is a problem. But people are quite powerful. Directly search for and modify the code for Godot to complete the repair.
Emmm.. The management can now close this question.. I'm not sure if I'll close it myself. Can you see this reply.. This comes from online translation.

**Comment by akien-mga** [comment]
>  So, this is an issue with DotNet 8.0. What are the issues with using Dotnet8.0 in Godot.. I see that others have also given feedback on the system XXXX is a problem. But people are quite powerful. Directly search for and modify the code for Godot to complete the repair.

Can you link us to what solutions you used? Is it something we need to change in the engine?

DotNet 8.0 should be supported and work fine, so if it doesn't, it's a bug we need to fix.

**Comment by 2dogtwo** [comment]
> > 因此，这是一个问题，DotNet8.0. 有什么问题与使用Dotnet8.0在Godot..我看看其他人也给予的反馈的系统XXXX是一个问题。 但是，人们相当强大。 直接搜索和修改的代码Godot完成的修复。
> 
> 你可以链接我们什么解决方案使用？ 这是我们需要改变的动机?
> 
> DotNet8.0应该支持和工作的现，因此，如果不是，这是一个错误，我们需要修复。

I'm not sure if this is a problem with Godot. Moreover, the platform I use for Godot is the Debian arm64 system of the chroot container. The system is inherently incomplete. So there was a malfunction. I can't tell whose problem it is. I only know now that I will delete net8.0. Both net6.0 and net7.9.0 allow me to compile a C # version of Godot that runs normally. I just tested version 4.3 (compiled a few days ago), which I previously compiled and ran with errors. It can also run normally under net7.0 and net6.0, so I found out that the problem is with net8.0. However, a few days ago, I was searching for a solution. See other feedback with similar issues. So I suspect it's also caused by Net8.0. But this can no longer be argued for. Because that feedback has already been resolved. It seems to have been resolved by modifying some code in Godot. So. I just mentioned whether Net8.0 itself is incomplete and has poor compatibility, or if there are any operations that can easily cause problems when using Net8.0 in Godot, but I cannot verify these on my own. I am just a beginner in programming. I don't know C++either. Emmm. By the way. The issue caused by net8.0 is not the inability to export Android APK. But it is unable to run and compile C # projects. I don't know if the exported compiled project can run. But it cannot run inside the editor. Even, you don't need any action. Just loaded the project. Wait for a moment, the editor will crash. But if you create a new one that doesn't use c #. It's just a GD project. Godot won't have any problems either. However, version 4.3 of Godot compilation requires Dotnet8.0. And I also successfully compiled it. This should indicate that my system environment should be able to run net8.0. I have also tested the removal of the Android SDK, only in the case of Godot and Net8.0. The problem still persists. So the problem lies in one of them.
